dbms/src/Storages/DeltaMerge/Delta/ColumnFilePersistedSet.cpp

```diff
@@ -27,70 +27,52 @@ namespace DB
+inline void serializeColumnFilePersisteds(WriteBatches & wbs, PageId id, const ColumnFilePersisteds & persisted_files)
 {
     MemoryWriteBuffer buf(0, COLUMN_FILE_SERIALIZE_BUFFER_SIZE);
-    auto column_files = flattenColumnFileLevels(file_levels);
-    serializeSavedColumnFiles(buf, column_files);
+    serializeSavedColumnFiles(buf, persisted_files);
     auto data_size = buf.count();
     wbs.meta.putPage(id, 0, buf.tryGetReadBuffer(), data_size);
 }
 
 void ColumnFilePersistedSet::updateColumnFileStats()
 {
-    size_t new_persisted_files_count = 0;
     size_t new_rows = 0;
     size_t new_bytes = 0;
     size_t new_deletes = 0;
-    for (auto & file_level : persisted_files_levels)
+    for (auto & file : persisted_files)
     {
-        new_persisted_files_count += file_level.size();
-        for (auto & file : file_level)
-        {
-            new_rows += file->getRows();
-            new_bytes += file->getBytes();
-            new_deletes += file->getDeletes();
-        }
+        new_rows += file->getRows();
+        new_bytes += file->getBytes();
+        new_deletes += file->getDeletes();
     }
-    persisted_files_count = new_persisted_files_count;
-    persisted_files_level_count = persisted_files_levels.size();
+    persisted_files_count = persisted_files.size();
     rows = new_rows;
     bytes = new_bytes;
     deletes = new_deletes;
 }
 
-void ColumnFilePersistedSet::checkColumnFiles(const ColumnFilePersistedLevels & new_column_file_levels)
+void ColumnFilePersistedSet::checkColumnFiles(const ColumnFilePersisteds & new_column_files)
 {
     if constexpr (!DM_RUN_CHECK)
         return;
     size_t new_rows = 0;
     size_t new_deletes = 0;
-    for (const auto & level : new_column_file_levels)
+    for (const auto & file : new_column_files)
     {
-        for (const auto & file : level)
-        {
-            new_rows += file->getRows();
-            new_deletes += file->isDeleteRange();
-        }
+        new_rows += file->getRows();
+        new_deletes += file->isDeleteRange();
     }
 
     if (unlikely(new_rows != rows || new_deletes != deletes))
     {
-        LOG_ERROR(log, "Rows and deletes check failed. Actual: rows[{}], deletes[{}]. Expected: rows[{}], deletes[{}]. Current column files: {}, new column files: {}.", new_rows, new_deletes, rows.load(), deletes.load(), columnFilesToString(flattenColumnFileLevels(persisted_files_levels)), columnFilesToString(flattenColumnFileLevels(new_column_file_levels)));
+        LOG_ERROR(log, "Rows and deletes check failed. Actual: rows[{}], deletes[{}]. Expected: rows[{}], deletes[{}]. Current column files: {}, new column files: {}.", //
+                  new_rows,
+                  new_deletes,
+                  rows.load(),
+                  deletes.load(),
+                  columnFilesToString(persisted_files),
+                  columnFilesToString(new_column_files));
         throw Exception("Rows and deletes check failed.", ErrorCodes::LOGICAL_ERROR);
     }
```

Comment:
not important: We can use RUNTIME_CHECK to replace these rows

Reply:
Fixed.